agent/ae/ae_test.go

```diff
@@ -27,50 +25,57 @@ func TestAE_scaleFactor(t *testing.T) {
 		{1000, 4},
 		{10000, 8},
 	}
-	for _, tt := range tests {
-		t.Run(fmt.Sprintf("%d nodes", tt.nodes), func(t *testing.T) {
-			if got, want := scaleFactor(tt.nodes), tt.scale; got != want {
-				t.Fatalf("got scale factor %d want %d", got, want)
-			}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("%d nodes", tc.nodes), func(t *testing.T) {
+			require.Equal(t, tc.scale, scaleFactor(tc.nodes))
 		})
 	}
 }
 
-func TestAE_Pause_nestedPauseResume(t *testing.T) {
-	t.Parallel()
-	l := NewStateSyncer(nil, 0, nil, nil)
-	if l.isPaused() != false {
-		t.Fatal("syncer should be unPaused after init")
-	}
-	l.Pause()
-	if l.isPaused() != true {
-		t.Fatal("syncer should be Paused after first call to Pause()")
-	}
-	l.Pause()
-	if l.isPaused() != true {
-		t.Fatal("syncer should STILL be Paused after second call to Pause()")
-	}
-	gotR := l.Resume()
-	if l.isPaused() != true {
-		t.Fatal("syncer should STILL be Paused after FIRST call to Resume()")
-	}
-	assert.False(t, gotR)
-	gotR = l.Resume()
-	if l.isPaused() != false {
-		t.Fatal("syncer should NOT be Paused after SECOND call to Resume()")
+func TestStateSyncer_Pause_nestedPauseResume(t *testing.T) {
+	sync := NewStateSyncer(nil, 0, nil, nil)
+
+	require.False(t, sync.isPaused(), "syncer should be unPaused after init")
+	require.Nil(t, sync.WaitResume())
+
+	sync.Pause()
+	require.True(t, sync.isPaused(), "syncer should be Paused after first call to Pause()")
+	isBlocked(t, sync.WaitResume())
+
+	sync.Pause()
+	require.True(t, sync.isPaused(), "syncer should STILL be Paused after second call to Pause()")
+	isBlocked(t, sync.WaitResume())
+
+	resumed := sync.Resume()
+	require.False(t, resumed)
+	require.True(t, sync.isPaused(), "syncer should STILL be Paused after FIRST call to Resume()")
+	chWaitResume := sync.WaitResume()
```

Comment:
I think this test should call `WaitResume` twice, to show that all watchers are notified when resume happens.